$NetBSD: patch-ai,v 1.30 2013/09/30 15:21:15 taca Exp $

Make this pkgsrc friendly.
Add support for NetBSD 5.x, NetBSD 6.x and DragonFly BSD.

--- makedefs.orig	2013-02-04 01:33:13.000000000 +0000
+++ makedefs
@@ -155,6 +155,8 @@ case "$SYSTEM.$RELEASE" in
 		;;
   FreeBSD.9*)	SYSTYPE=FREEBSD9
 		;;
+  DragonFly.*)	SYSTYPE=DRAGONFLY
+		;;
   OpenBSD.2*)	SYSTYPE=OPENBSD2
 		;;
   OpenBSD.3*)	SYSTYPE=OPENBSD3
@@ -173,6 +175,10 @@ case "$SYSTEM.$RELEASE" in
 		;;
    NetBSD.4*)	SYSTYPE=NETBSD4
 		;;
+   NetBSD.5*)	SYSTYPE=NETBSD5
+		;;
+   NetBSD.6*)	SYSTYPE=NETBSD6
+		;;
    BSD/OS.2*)	SYSTYPE=BSDI2
 		;;
    BSD/OS.3*)	SYSTYPE=BSDI3
@@ -226,13 +232,6 @@ case "$SYSTEM.$RELEASE" in
 		esac
 		;;
    ULTRIX.4*)	SYSTYPE=ULTRIX4
-		if [ -f /usr/local/lib/libdb.a ]; then
-		    SYSLIBS="$SYSLIBS -ldb"
-		    CCARGS="$CCARGS -DHAS_DB"
-		    if [ -d /usr/local/include/db ]; then
-			CCARGS="$CCARGS -I/usr/local/include/db"
-		    fi
-		fi
 		for l in syslog resolv; do
 		    if [ -f /usr/local/lib/lib$l.a ]; then
 			SYSLIBS="$SYSLIBS -l$l"
@@ -270,31 +269,8 @@ case "$SYSTEM.$RELEASE" in
 		esac;;
 		# Tested with RedHat 3.03 on 20020729.
     Linux.1*)	SYSTYPE=LINUX1
-		case "$CCARGS" in
-		 *-DNO_DB*) ;;
-		 *) SYSLIBS="-ldb";;
-		esac
 		;;
     Linux.2*)	SYSTYPE=LINUX2
-		case "$CCARGS" in
-		 *-DNO_DB*) ;;
-		 *) if [ -f /usr/include/db.h ]
-		    then
-			: we are all set
-		    elif [ -f /usr/include/db/db.h ]
-		    then
-			CCARGS="$CCARGS -I/usr/include/db"
-		    else
-			# No, we're not going to try db1 db2 db3 etc.
-			# On a properly installed system, Postfix builds
-			# by including <db.h> and by linking with -ldb
-			echo "No <db.h> include file found." 1>&2
-			echo "Install the appropriate db*-devel package first." 1>&2
-			exit 1
-		    fi
-		    SYSLIBS="-ldb"
-		    ;;
-		esac
 		for name in nsl resolv $GDBM_LIBS
 		do
 		    for lib in /usr/lib64 /lib64 /usr/lib /lib
@@ -348,24 +324,6 @@ EOF
 		esac
 		;;
     Linux.3*)	SYSTYPE=LINUX3
-		case "$CCARGS" in
-		 *-DNO_DB*) ;;
-		 *) if [ -f /usr/include/db.h ]
-		    then
-			: we are all set
-		    elif [ -f /usr/include/db/db.h ]
-		    then
-			CCARGS="$CCARGS -I/usr/include/db"
-		    else
-			# On a properly installed system, Postfix builds
-			# by including <db.h> and by linking with -ldb
-			echo "No <db.h> include file found." 1>&2
-			echo "Install the appropriate db*-devel package first." 1>&2
-			exit 1
-		    fi
-		    SYSLIBS="-ldb"
-		    ;;
-		esac
 		for name in nsl resolv
 		do
 		    for lib in /usr/lib64 /lib64 /usr/lib /usr/lib/* /lib /lib/*
@@ -379,24 +337,6 @@ EOF
 		;;
      GNU.0*|GNU/kFreeBSD.[567]*)
      		SYSTYPE=GNU0
-		case "$CCARGS" in
-		 *-DNO_DB*) ;;
-		 *) if [ -f /usr/include/db.h ]
-		    then
-			: we are all set
-		    elif [ -f /usr/include/db/db.h ]
-		    then
-			CCARGS="$CCARGS -I/usr/include/db"
-		    else
-			# On a properly installed system, Postfix builds
-			# by including <db.h> and by linking with -ldb
-			echo "No <db.h> include file found." 1>&2
-			echo "Install the appropriate db*-devel package first." 1>&2
-			exit 1
-		    fi
-		    SYSLIBS="-ldb"
-		    ;;
-		esac
 		for name in nsl resolv
 		do
 		    for lib in /usr/lib64 /lib64 /usr/lib /lib
@@ -427,25 +367,13 @@ EOF
 HP-UX.A.09.*)	SYSTYPE=HPUX9
 		SYSLIBS=-ldbm
 		CCARGS="$CCARGS -DMISSING_USLEEP"
-		if [ -f /usr/lib/libdb.a ]; then
-		    CCARGS="$CCARGS -DHAS_DB"
-		    SYSLIBS="$SYSLIBS -ldb"
-		fi
 		;;
 HP-UX.B.10.*)	SYSTYPE=HPUX10
 		CCARGS="$CCARGS `nm /usr/lib/libc.a 2>/dev/null |
 		    (grep usleep >/dev/null || echo '-DMISSING_USLEEP')`"
-		if [ -f /usr/lib/libdb.a ]; then
-		    CCARGS="$CCARGS -DHAS_DB"
-		    SYSLIBS=-ldb
-		fi
 		;;
 HP-UX.B.11.*)	SYSTYPE=HPUX11
 		SYSLIBS=-lnsl
-		if [ -f /usr/lib/libdb.a ]; then
-		    CCARGS="$CCARGS -DHAS_DB"
-		    SYSLIBS="$SYSLIBS -ldb"
-		fi
 		;;
 ReliantUNIX-?.5.43) SYSTYPE=ReliantUnix543
 		RANLIB=echo
